src/NUnitEngine/Addins/addin-tests/VisualStudioProjectLoaderTests.cs

```diff
@@ -56,6 +56,45 @@ public void ValidExtensions(string project, bool isGood)
         }
 
         [Test]
+        public void CheckExtensionAttribute()
+        {
+            Assert.That(typeof(VisualStudioProjectLoader),
+                Has.Attribute<ExtensionAttribute>());
+        }
+
+        // Note for review:
+        // The following test doesn't pass because AttributeConstraint always uses the
+        // first attribute it finds. Should we fix the syntax or just document it?
```

Comment:
I will. I'm not 100% sure we can fix it - I know it's not an easy one. The evaluation proceeds from left to right and the attribute we use is selected by name when we hit HasAttribute. What we want to do instead is select all the attributes and test each of them in (logical) parallel to see if one passes.